diff -ur xen-4.16.2.orig/xen/common/bunzip2.c xen-4.16.2/xen/common/bunzip2.c
--- xen-4.16.2.orig/xen/common/bunzip2.c	2022-08-18 13:47:46.000000000 +0200
+++ xen-4.16.2/xen/common/bunzip2.c	2023-07-20 22:01:47.934242320 +0200
@@ -223,6 +223,7 @@
 	for (j = 0; j < groupCount; j++) {
 		unsigned char length[MAX_SYMBOLS], temp[MAX_HUFCODE_BITS+1];
 		int	minLen,	maxLen, pp;
+		length[0] = 0;
 		/* Read Huffman code lengths for each symbol.  They're
 		   stored in a way similar to mtf; record a starting
 		   value for the first symbol, and an offset from the
