diff -urN xen-4.6.5/tools/libxl/libxl_internal.h xen-4.6.5.build/tools/libxl/libxl_internal.h
--- xen-4.6.5/tools/libxl/libxl_internal.h	2017-03-07 17:19:05.000000000 +0100
+++ xen-4.6.5.build/tools/libxl/libxl_internal.h	2017-04-14 15:05:55.088806418 +0200
@@ -37,6 +37,7 @@
 #include <unistd.h>
 #include <ctype.h>
 
+#include <sys/sysmacros.h>
 #include <sys/mman.h>
 #include <sys/poll.h>
 #include <sys/select.h>
