diff -urNp -x '*.orig' xen-4.13.1.org/tools/libxl/libxl_internal.h xen-4.13.1/tools/libxl/libxl_internal.h
--- xen-4.13.1.org/tools/libxl/libxl_internal.h	2020-05-14 14:19:32.000000000 +0200
+++ xen-4.13.1/tools/libxl/libxl_internal.h	2021-03-25 22:42:35.034240136 +0100
@@ -39,6 +39,7 @@
 #include <unistd.h>
 #include <ctype.h>
 
+#include <sys/sysmacros.h>
 #include <sys/mman.h>
 #include <poll.h>
 #include <sys/select.h>
