diff -dur -x '*~' xen-3.0.2.orig/tools/examples/network-bridge xen-3.0.2/tools/examples/network-bridge
--- xen-3.0.2.orig/tools/examples/network-bridge	2006-06-29 14:58:24.000000000 +0200
+++ xen-3.0.2/tools/examples/network-bridge	2006-06-29 14:59:49.000000000 +0200
@@ -61,7 +61,7 @@
 
 vifnum=${vifnum:-0}
 bridge=${bridge:-xenbr${vifnum}}
-netdev=${netdev:-$(ip route list default scope global| awk '{ print $NF }')}
+netdev=${netdev:-$(ip route list default scope global | sed -e's/ onlink//' | awk '{ print $NF }')}
 netdev=${netdev:-eth${vifnum}}
 antispoof=${antispoof:-no}
 
