--- xen-4.18.1/tools/hotplug/Linux/systemd/xenconsoled.service.in.orig	2024-04-22 20:04:39.052416625 +0200
+++ xen-4.18.1/tools/hotplug/Linux/systemd/xenconsoled.service.in	2024-04-22 20:06:13.978569033 +0200
@@ -7,12 +7,12 @@ ConditionPathExists=/proc/xen/capabiliti
 [Service]
 Type=simple
 Environment=XENCONSOLED_ARGS=
-Environment=XENCONSOLED_TRACE=none
+Environment=XENCONSOLED_LOG=none
 Environment=XENCONSOLED_LOG_DIR=@XEN_LOG_DIR@/console
-EnvironmentFile=-@CONFIG_DIR@/@CONFIG_LEAF_DIR@/xencommons
+EnvironmentFile=-@CONFIG_DIR@/@CONFIG_LEAF_DIR@/xenconsoled
 ExecStartPre=/bin/grep -q control_d /proc/xen/capabilities
 ExecStartPre=/bin/mkdir -p ${XENCONSOLED_LOG_DIR} @XEN_RUN_DIR@
-ExecStart=@sbindir@/xenconsoled -i --log=${XENCONSOLED_TRACE} --log-dir=${XENCONSOLED_LOG_DIR} $XENCONSOLED_ARGS
+ExecStart=@sbindir@/xenconsoled -i --log=${XENCONSOLED_LOG} --log-dir=${XENCONSOLED_LOG_DIR} $XENCONSOLED_ARGS
 
 [Install]
 WantedBy=multi-user.target
--- xen-4.9.0/tools/hotplug/Linux/launch-xenstore.in.orig	2017-06-27 20:13:19.000000000 +0200
+++ xen-4.9.0/tools/hotplug/Linux/launch-xenstore.in	2017-07-30 21:16:13.246222287 +0200
@@ -46,7 +46,7 @@
 
 test_xenstore && exit 0
 
-test -f @CONFIG_DIR@/@CONFIG_LEAF_DIR@/xencommons && . @CONFIG_DIR@/@CONFIG_LEAF_DIR@/xencommons
+test -f @CONFIG_DIR@/@CONFIG_LEAF_DIR@/xenstored && . @CONFIG_DIR@/@CONFIG_LEAF_DIR@/xenstored
 
 [ "$XENSTORETYPE" = "" ] && XENSTORETYPE=daemon
 
