--- xen-3.0.3_0-src/tools/pygrub/src/fsys/reiser/reisermodule.c	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/pygrub/src/fsys/reiser/reisermodule.c-new	2006-11-24 01:31:53.058527280 +0100
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include <dal/file_dal.h>
+#include <dal/file.h>
 #include <reiserfs/reiserfs.h>
 
 #if (PYTHON_API_VERSION >= 1011)
@@ -42,12 +42,12 @@
     reiserfs_file_t *file;
 };
 
-void file_dal_close(dal_t *dal) {
+void file_close(dal_t *dal) {
 
 	if (!dal) return;
 
-	close((int)(unsigned long)dal->dev);
-	dal_free(dal);
+	close((int)(unsigned long)dal->entity);
+//	dal_free(dal);
 }
 
 /* reiser file object */
@@ -195,7 +195,7 @@
     if (fs->fs != NULL)
     {
         reiserfs_fs_close(fs->fs);
-        file_dal_close(fs->dal);
+        file_close(fs->dal);
 		fs->fs = NULL;
     }
     Py_INCREF(Py_None);
@@ -218,13 +218,13 @@
         return NULL;
     }
 
-    if (!(dal = file_dal_open(name, block_size, O_RDONLY))) {
+    if (!(dal = file_open(name, block_size, O_RDONLY))) {
         PyErr_SetString(PyExc_ValueError, "Couldn't create device abstraction");
         return NULL;    
     }
     
     if (!(rfs = reiserfs_fs_open_fast(dal, dal))) {
-		file_dal_close(dal);
+		file_close(dal);
         PyErr_SetString(PyExc_ValueError, "unable to open file");
         return NULL;
     }
@@ -265,7 +265,7 @@
     if (fs->fs != NULL)
 	{
         reiserfs_fs_close(fs->fs);
-		file_dal_close(fs->dal);
+		file_close(fs->dal);
 		fs->fs = NULL;
 	}
 	PyObject_DEL(fs);
