diff -up xen-3.4.0/tools/Makefile.dump xen-3.4.0/tools/Makefile
--- xen-3.4.0/tools/Makefile.dump	2009-05-18 13:05:38.000000000 +0200
+++ xen-3.4.0/tools/Makefile	2009-05-20 17:03:26.000000000 +0200
@@ -46,7 +46,7 @@ all: subdirs-all
 
 .PHONY: install
 install: subdirs-install
-	$(INSTALL_DIR) $(DESTDIR)/var/xen/dump
+	$(INSTALL_DIR) $(DESTDIR)/var/lib/xen/dump
 	$(INSTALL_DIR) $(DESTDIR)/var/log/xen
 	$(INSTALL_DIR) $(DESTDIR)/var/lib/xen
 	$(INSTALL_DIR) $(DESTDIR)/var/lock/subsys
--- xen-4.5.1/tools/libxl/xl_cmdimpl.c.orig	2015-06-22 15:41:35.000000000 +0200
+++ xen-4.5.1/tools/libxl/xl_cmdimpl.c	2015-10-01 17:45:14.267331067 +0200
@@ -2092,7 +2092,7 @@
         char *corefile;
         int rc;
 
-        if (asprintf(&corefile, "/var/xen/dump/%s", d_config->c_info.name) < 0) {
+        if (asprintf(&corefile, "/var/lib/xen/dump/%s", d_config->c_info.name) < 0) {
             LOG("failed to construct core dump path");
         } else {
             LOG("dumping core to %s", corefile);
--- xen-4.5.1/docs/man/xl.cfg.pod.5.orig	2015-06-22 15:41:35.000000000 +0200
+++ xen-4.5.1/docs/man/xl.cfg.pod.5	2015-10-01 17:46:38.750661228 +0200
@@ -298,12 +298,12 @@
 
 =item B<coredump-destroy>
 
-write a "coredump" of the domain to F</var/xen/dump/NAME> and then
+write a "coredump" of the domain to F</var/lib/xen/dump/NAME> and then
 destroy the domain.
 
 =item B<coredump-restart>
 
-write a "coredump" of the domain to F</var/xen/dump/NAME> and then
+write a "coredump" of the domain to F</var/lib/xen/dump/NAME> and then
 restart the domain.
 
 =back
@@ -1605,7 +1605,7 @@
 =head1 FILES
 
 F</etc/xen/NAME.cfg>
-F</var/xen/dump/NAME>
+F</var/lib/xen/dump/NAME>
 
 =head1 BUGS
 
