diff -urN xen-unstable.org/tools/pygrub/src/fsys/reiser/reisermodule.c xen-unstable/tools/pygrub/src/fsys/reiser/reisermodule.c
--- xen-unstable.org/tools/pygrub/src/fsys/reiser/reisermodule.c	2006-01-25 05:56:11.000000000 +0100
+++ xen-unstable/tools/pygrub/src/fsys/reiser/reisermodule.c	2006-01-25 18:22:47.110770500 +0100
@@ -17,7 +17,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-#include <dal/file_dal.h>
+#include <dal/file.h>
 #include <reiserfs/reiserfs.h>
 
 #if (PYTHON_API_VERSION >= 1011)
@@ -45,9 +45,7 @@
 void file_dal_close(dal_t *dal) {
 
 	if (!dal) return;
-
-	close((int)(unsigned long)dal->dev);
-	dal_free(dal);
+	file_close(dal);
 }
 
 /* reiser file object */
@@ -218,7 +216,7 @@
         return NULL;
     }
 
-    if (!(dal = file_dal_open(name, block_size, O_RDONLY))) {
+    if (!(dal = file_open(name, block_size, O_RDONLY))) {
         PyErr_SetString(PyExc_ValueError, "Couldn't create device abstraction");
         return NULL;    
     }
