--- xen-4.15.2/tools/libs/uselibs.mk.orig	2022-01-31 10:42:09.000000000 +0100
+++ xen-4.15.2/tools/libs/uselibs.mk	2022-06-11 17:42:08.053614271 +0200
@@ -19,7 +19,7 @@ USELIBS_hypfs := toollog toolcore call
 LIBS_LIBS += ctrl
 USELIBS_ctrl := toollog call evtchn gnttab foreignmemory devicemodel
 LIBS_LIBS += guest
-USELIBS_guest := evtchn ctrl
+USELIBS_guest := evtchn ctrl toollog call foreignmemory
 LIBS_LIBS += store
 USELIBS_store := toolcore
 LIBS_LIBS += vchan
