diff -urN xen-3.0.3_0-src.orig/tools/check/check_logging xen-3.0.3_0-src/tools/check/check_logging
--- xen-3.0.3_0-src.orig/tools/check/check_logging	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/check/check_logging	2006-10-30 15:26:46.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #  -*- mode: python; -*-
 
 import os
diff -urN xen-3.0.3_0-src.orig/tools/misc/netfix xen-3.0.3_0-src/tools/misc/netfix
--- xen-3.0.3_0-src.orig/tools/misc/netfix	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/netfix	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #  -*- mode: python; -*-
 #============================================================================
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
diff -urN xen-3.0.3_0-src.orig/tools/misc/xen-bugtool xen-3.0.3_0-src/tools/misc/xen-bugtool
--- xen-3.0.3_0-src.orig/tools/misc/xen-bugtool	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/xen-bugtool	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 #  -*- mode: python; -*-
 
diff -urN xen-3.0.3_0-src.orig/tools/misc/xencons xen-3.0.3_0-src/tools/misc/xencons
--- xen-3.0.3_0-src.orig/tools/misc/xencons	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/xencons	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 ##############################################
 # Console client for Xen guest OSes
diff -urN xen-3.0.3_0-src.orig/tools/misc/xend xen-3.0.3_0-src/tools/misc/xend
--- xen-3.0.3_0-src.orig/tools/misc/xend	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/xend	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #  -*- mode: python; -*-
 #============================================================================
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
diff -urN xen-3.0.3_0-src.orig/tools/misc/xensymoops xen-3.0.3_0-src/tools/misc/xensymoops
--- xen-3.0.3_0-src.orig/tools/misc/xensymoops	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/xensymoops	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # An oops analyser for Xen
 # Usage: xensymoops path-to-xen.s < oops-message
diff -urN xen-3.0.3_0-src.orig/tools/misc/xm xen-3.0.3_0-src/tools/misc/xm
--- xen-3.0.3_0-src.orig/tools/misc/xm	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/misc/xm	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #  -*- mode: python; -*-
 import sys
 
diff -urN xen-3.0.3_0-src.orig/tools/python/xen/util/bugtool.py xen-3.0.3_0-src/tools/python/xen/util/bugtool.py
--- xen-3.0.3_0-src.orig/tools/python/xen/util/bugtool.py	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/python/xen/util/bugtool.py	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
diff -urN xen-3.0.3_0-src.orig/tools/python/xen/util/pci.py xen-3.0.3_0-src/tools/python/xen/util/pci.py
--- xen-3.0.3_0-src.orig/tools/python/xen/util/pci.py	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/python/xen/util/pci.py	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #
 # PCI Device Information Class
 # - Helps obtain information about which I/O resources a PCI device needs
diff -urN xen-3.0.3_0-src.orig/tools/python/xen/xend/sxp.py xen-3.0.3_0-src/tools/python/xen/xend/sxp.py
--- xen-3.0.3_0-src.orig/tools/python/xen/xend/sxp.py	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/python/xen/xend/sxp.py	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
diff -urN xen-3.0.3_0-src.orig/tools/python/xen/xend/XendClient.py xen-3.0.3_0-src/tools/python/xen/xend/XendClient.py
--- xen-3.0.3_0-src.orig/tools/python/xen/xend/XendClient.py	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/python/xen/xend/XendClient.py	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #============================================================================
 # This library is free software; you can redistribute it and/or
 # modify it under the terms of version 2.1 of the GNU Lesser General Public
diff -urN xen-3.0.3_0-src.orig/tools/vnet/scripts/vn xen-3.0.3_0-src/tools/vnet/scripts/vn
--- xen-3.0.3_0-src.orig/tools/vnet/scripts/vn	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/vnet/scripts/vn	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.4
+#!/usr/bin/python
 #  -*- mode: python; -*-
 #============================================================================
 # Copyright (C) 2005, 2006 Mike Wray <mike.wray@hp.com>
diff -urN xen-3.0.3_0-src.orig/tools/xenmon/xenmon.py xen-3.0.3_0-src/tools/xenmon/xenmon.py
--- xen-3.0.3_0-src.orig/tools/xenmon/xenmon.py	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/xenmon/xenmon.py	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 #####################################################################
 # xenmon is a front-end for xenbaked.
diff -urN xen-3.0.3_0-src.orig/tools/xentrace/xentrace_format xen-3.0.3_0-src/tools/xentrace/xentrace_format
--- xen-3.0.3_0-src.orig/tools/xentrace/xentrace_format	2006-10-15 14:22:03.000000000 +0200
+++ xen-3.0.3_0-src/tools/xentrace/xentrace_format	2006-10-30 15:26:47.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 
 # by Mark Williamson, (C) 2004 Intel Research Cambridge
 
--- xen-3.0.4_1-src/tools/check/check_python_devel	2007-01-08 15:00:47.000000000 +0000
+++ xen-3.0.4_1-src/tools/check/check_python_devel	2007-01-10 10:15:10.000000000 +0000
@@ -5,7 +5,7 @@ RC=0
 
 python -c '
 import os.path, sys
 for p in sys.path:
-	if os.path.exists(p + "/config/Makefile"):
+	if os.path.exists(p + "/../config/Makefile"):
 		sys.exit(0)
 sys.exit(1)
