--- xen-4.2.0.orig/tools/xenstat/Makefile	2012-09-17 12:21:19.000000000 +0200
+++ xen-4.2.0/tools/xenstat/Makefile	2012-10-23 09:59:58.000000000 +0200
@@ -6,7 +6,7 @@
 
 # This doesn't cross-compile (cross-compile environments rarely have curses)
 ifeq ($(XEN_COMPILE_ARCH),$(XEN_TARGET_ARCH))
-ifeq ($(wildcard /usr/include/curses.h),/usr/include/curses.h)
+ifeq ($(wildcard /usr/include/ncurses/curses.h),/usr/include/ncurses/curses.h)
 SUBDIRS += xentop
 endif
 endif
