--- xen-3.0.4_1-src/xen/common/symbols.c~	2007-01-08 17:00:54.000000000 +0200
+++ xen-3.0.4_1-src/xen/common/symbols.c	2008-05-24 01:03:53.860604056 +0300
@@ -94,7 +94,7 @@
     unsigned long symbol_end = 0;
 
     /* This kernel should never had been booted. */
-    BUG_ON(!symbols_addresses);
+//    BUG_ON(!symbols_addresses);
 
     namebuf[KSYM_NAME_LEN] = 0;
     namebuf[0] = 0;
--- xen-3.0.4_1-src/Config.mk~	2008-05-24 01:49:20.000000000 +0300
+++ xen-3.0.4_1-src/Config.mk	2008-05-24 01:49:23.342299929 +0300
@@ -48,6 +48,9 @@
 HOSTCFLAGS += $(call cc-option,$(HOSTCC),-Wdeclaration-after-statement,)
 CFLAGS     += $(call cc-option,$(CC),-Wdeclaration-after-statement,)
 
+HOSTCFLAGS += $(call cc-option,$(HOSTCC),-fgnu89-inline,)
+CFLAGS     += $(call cc-option,$(CC),-fgnu89-inline,)
+
 LDFLAGS += $(foreach i, $(EXTRA_LIB), -L$(i)) 
 CFLAGS += $(foreach i, $(EXTRA_INCLUDES), -I$(i))
 
