diff -urNp -x '*.orig' xen-4.13.1.org/tools/qemu-xen-traditional/Makefile xen-4.13.1/tools/qemu-xen-traditional/Makefile
--- xen-4.13.1.org/tools/qemu-xen-traditional/Makefile	2018-10-24 17:18:11.000000000 +0200
+++ xen-4.13.1/tools/qemu-xen-traditional/Makefile	2021-03-25 22:42:21.257653071 +0100
@@ -275,7 +275,7 @@ cscope:
 
 # documentation
 %.html: %.texi
-	texi2html -monolithic -number $<
+	texi2any --html --no-split --number-sections $<
 
 %.info: %.texi
 	makeinfo $< -o $@
diff -urNp -x '*.orig' xen-4.13.1.org/tools/qemu-xen-traditional/configure xen-4.13.1/tools/qemu-xen-traditional/configure
--- xen-4.13.1.org/tools/qemu-xen-traditional/configure	2021-03-25 22:42:20.504324047 +0100
+++ xen-4.13.1/tools/qemu-xen-traditional/configure	2021-03-25 22:42:21.257653071 +0100
@@ -1087,7 +1087,7 @@ EOF
 fi
 
 # Check if tools are available to build documentation.
-if [ -x "`which texi2html 2>/dev/null`" ] && \
+if [ -x "`which texi2any 2>/dev/null`" ] && \
    [ -x "`which pod2man 2>/dev/null`" ]; then
   build_docs="yes"
 fi
diff -urNp -x '*.orig' xen-4.13.1.org/tools/qemu-xen-traditional/texi2pod.pl xen-4.13.1/tools/qemu-xen-traditional/texi2pod.pl
--- xen-4.13.1.org/tools/qemu-xen-traditional/texi2pod.pl	2018-10-24 17:18:11.000000000 +0200
+++ xen-4.13.1/tools/qemu-xen-traditional/texi2pod.pl	2021-03-25 22:42:21.257653071 +0100
@@ -340,6 +340,8 @@ die "No filename or title\n" unless defi
 $sects{NAME} = "$fn \- $tl\n";
 $sects{FOOTNOTES} .= "=back\n" if exists $sects{FOOTNOTES};
 
+print "=encoding utf8\n";
+
 for $sect (qw(NAME SYNOPSIS DESCRIPTION OPTIONS ENVIRONMENT FILES
 	      BUGS NOTES FOOTNOTES SEEALSO AUTHOR COPYRIGHT)) {
     if(exists $sects{$sect}) {
