--- xen-4.9.0/Config.mk.orig	2017-07-30 20:40:20.116246872 +0200
+++ xen-4.9.0/Config.mk	2017-07-30 20:43:13.782911554 +0200
@@ -29,7 +29,7 @@
 SHELL     ?= /bin/sh
 
 # Tools to run on system hosting the build
-HOSTCFLAGS  = -Wall -Werror -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS  = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer
 HOSTCFLAGS += -fno-strict-aliasing
 
 DISTDIR     ?= $(XEN_ROOT)/dist
