diff -urNp -x '*.orig' xen-4.13.1.org/tools/Makefile xen-4.13.1/tools/Makefile
--- xen-4.13.1.org/tools/Makefile	2020-05-14 14:19:32.000000000 +0200
+++ xen-4.13.1/tools/Makefile	2021-03-25 22:42:23.177642067 +0100
@@ -243,7 +243,7 @@ subdir-all-qemu-xen-dir: qemu-xen-dir-fi
 		$(QEMU_XEN_ENABLE_DEBUG) \
 		$$enable_trace_backend \
 		--prefix=$(LIBEXEC) \
-		--libdir=$(LIBEXEC_LIB) \
+		--libdir=$(LIBEXEC_LIB) --libexecdir=$(LIBEXEC_LIB) \
 		--includedir=$(LIBEXEC_INC) \
 		--extra-cflags="-DXC_WANT_COMPAT_EVTCHN_API=1 \
 		-DXC_WANT_COMPAT_GNTTAB_API=1 \
