--- xen-4.5.1/tools/Makefile.orig	2015-10-03 17:22:11.716804023 +0200
+++ xen-4.5.1/tools/Makefile	2015-10-03 19:37:54.219795649 +0200
@@ -207,7 +207,7 @@
 	$$source/configure --enable-xen --target-list=i386-softmmu \
 		$(QEMU_XEN_ENABLE_DEBUG) \
 		--prefix=$(LIBEXEC) \
-		--libdir=$(LIBEXEC_LIB) \
+		--libdir=$(LIBEXEC_LIB) --libexecdir=$(LIBEXEC_LIB) \
 		--includedir=$(LIBEXEC_INC) \
 		--source-path=$$source \
 		--extra-cflags="-I$(XEN_ROOT)/tools/include \
