diff -dur -x '*~' xen-4.1.2.orig/Config.mk xen-4.1.2/Config.mk
--- xen-4.1.2.orig/Config.mk	2011-10-20 19:06:04.000000000 +0200
+++ xen-4.1.2/Config.mk	2012-09-20 08:44:53.000000000 +0200
@@ -14,7 +14,7 @@
 
 # Tools to run on system hosting the build
 HOSTCC      = gcc
-HOSTCFLAGS  = -Wall -Werror -Wstrict-prototypes -O2 -fomit-frame-pointer
+HOSTCFLAGS  = -Wall -Wstrict-prototypes -O2 -fomit-frame-pointer
 HOSTCFLAGS += -fno-strict-aliasing
 
 DISTDIR     ?= $(XEN_ROOT)/dist
diff -dur -x '*~' xen-4.1.2.orig/xen/arch/x86/Rules.mk xen-4.1.2/xen/arch/x86/Rules.mk
--- xen-4.1.2.orig/xen/arch/x86/Rules.mk	2011-10-20 19:05:47.000000000 +0200
+++ xen-4.1.2/xen/arch/x86/Rules.mk	2012-09-20 08:45:47.000000000 +0200
@@ -17,7 +17,7 @@
 endif
 
 CFLAGS += -fno-builtin -fno-common -Wredundant-decls
-CFLAGS += -iwithprefix include -Werror -Wno-pointer-arith -pipe
+CFLAGS += -iwithprefix include -Wno-pointer-arith -pipe
 CFLAGS += -I$(BASEDIR)/include 
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-generic
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-default
diff -dur -x '*~' xen-4.1.2.orig/xen/arch/x86/boot/build32.mk xen-4.1.2/xen/arch/x86/boot/build32.mk
--- xen-4.1.2.orig/xen/arch/x86/boot/build32.mk	2011-10-20 19:05:47.000000000 +0200
+++ xen-4.1.2/xen/arch/x86/boot/build32.mk	2012-09-20 08:45:59.000000000 +0200
@@ -4,7 +4,7 @@
 
 $(call cc-options-add,CFLAGS,CC,$(EMBEDDED_EXTRA_CFLAGS))
 
-CFLAGS += -Werror -fno-builtin -msoft-float
+CFLAGS += -fno-builtin -msoft-float
 
 # NB. awk invocation is a portable alternative to 'head -n -1'
 %.S: %.bin
--- xen-4.4.1/tools/blktap/drivers/Makefile~	2014-09-02 08:20:19.000000000 +0200
+++ xen-4.4.1/tools/blktap/drivers/Makefile	2014-12-13 19:19:17.070688454 +0100
@@ -4,7 +4,6 @@
 IBIN         = blktapctrl tapdisk
 QCOW_UTIL    = img2qcow qcow2raw qcow-create
 
-CFLAGS   += -Werror
 CFLAGS   += -Wno-unused
 CFLAGS   += -I../lib
 CFLAGS   += $(CFLAGS_libxenctrl)
