diff -Naur xen-2.0.orig/tools/python/xen/xend/server/SrvBase.py xen-2.0/tools/python/xen/xend/server/SrvBase.py
--- xen-2.0.orig/tools/python/xen/xend/server/SrvBase.py	2005-05-22 12:56:27.000000000 +0200
+++ xen-2.0/tools/python/xen/xend/server/SrvBase.py	2005-06-12 20:08:20.000000000 +0200
@@ -9,7 +9,7 @@
 
 from twisted.internet import defer
 from twisted.internet import reactor
-from twisted.protocols import http
+from twisted.web import http
 from twisted.web import error
 from twisted.web import resource
 from twisted.web import server
diff -Naur xen-2.0.orig/tools/python/xen/xend/server/SrvDir.py xen-2.0/tools/python/xen/xend/server/SrvDir.py
--- xen-2.0.orig/tools/python/xen/xend/server/SrvDir.py	2005-05-22 12:56:27.000000000 +0200
+++ xen-2.0/tools/python/xen/xend/server/SrvDir.py	2005-06-12 20:08:26.000000000 +0200
@@ -1,6 +1,6 @@
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
 
-from twisted.protocols import http
+from twisted.web import http
 from twisted.web import error
 
 from xen.xend import sxp
diff -Naur xen-2.0.orig/tools/python/xen/xend/server/SrvDomainDir.py xen-2.0/tools/python/xen/xend/server/SrvDomainDir.py
--- xen-2.0.orig/tools/python/xen/xend/server/SrvDomainDir.py	2005-05-22 12:56:31.000000000 +0200
+++ xen-2.0/tools/python/xen/xend/server/SrvDomainDir.py	2005-06-12 20:08:40.000000000 +0200
@@ -3,7 +3,7 @@
 import traceback
 from StringIO import StringIO
 
-from twisted.protocols import http
+from twisted.web import http
 from twisted.web import error
 from twisted.python.failure import Failure
 
diff -Naur xen-2.0.orig/tools/python/xen/xend/server/SrvDomain.py xen-2.0/tools/python/xen/xend/server/SrvDomain.py
--- xen-2.0.orig/tools/python/xen/xend/server/SrvDomain.py	2005-05-22 12:56:30.000000000 +0200
+++ xen-2.0/tools/python/xen/xend/server/SrvDomain.py	2005-06-12 20:08:33.000000000 +0200
@@ -1,6 +1,6 @@
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
 
-from twisted.protocols import http
+from twisted.web import http
 
 from xen.xend import sxp
 from xen.xend import XendDomain
diff -Naur xen-2.0.orig/tools/python/xen/xend/XendAsynchProtocol.py xen-2.0/tools/python/xen/xend/XendAsynchProtocol.py
--- xen-2.0.orig/tools/python/xen/xend/XendAsynchProtocol.py	2005-05-22 12:56:31.000000000 +0200
+++ xen-2.0/tools/python/xen/xend/XendAsynchProtocol.py	2005-06-12 20:08:10.000000000 +0200
@@ -1,6 +1,6 @@
 # Copyright (C) 2004 Mike Wray <mike.wray@hp.com>
 
-from twisted.protocols import http
+from twisted.web import http
 from twisted.internet.protocol import ClientCreator
 from twisted.internet.defer import Deferred
 from twisted.internet import reactor
