--- xen-4.15.3/Config.mk.orig	2022-09-18 21:17:34.143899962 +0200
+++ xen-4.15.3/Config.mk	2022-09-18 21:34:45.351065045 +0200
@@ -192,6 +192,8 @@ $(call cc-option-add,HOSTCFLAGS,HOSTCC,-
 $(call cc-option-add,CFLAGS,CC,-Wdeclaration-after-statement)
 $(call cc-option-add,CFLAGS,CC,-Wno-unused-but-set-variable)
 $(call cc-option-add,CFLAGS,CC,-Wno-unused-local-typedefs)
+$(call cc-option-add,CFLAGS,CC,-Wno-error=array-bounds)
+$(call cc-option-add,CFLAGS,CC,-Wno-error=address)
 
 LDFLAGS += $(foreach i, $(EXTRA_LIB), -L$(i)) 
 CFLAGS += $(foreach i, $(EXTRA_INCLUDES), -I$(i))
