--- xen-2.0/tools/examples/network.orig	2005-05-22 12:56:31.000000000 +0200
+++ xen-2.0/tools/examples/network	2005-06-13 14:23:26.000000000 +0200
@@ -69,7 +69,7 @@
     # to 'dev $src'. Remove netmask as we'll add routes later.
     ip addr show dev ${src} | egrep '^ *inet ' | sed -e "
 s/inet/ip addr add/
-s@\([0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+\)/[0-9]\+@\1@
+s@\([0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+\)/[0-9]\*@\1@
 s/${src}/dev ${dst}/
 " | sh -e
 }
@@ -104,7 +104,7 @@
 s/^/ip route add /
 P
 d
-" | sh -e
+" | grep -v scope | sh -e 
 }
 
 # Usage: create_bridge bridge
@@ -185,6 +185,7 @@
     else
         transfer_addrs ${netdev} ${bridge}
         transfer_routes ${netdev} ${bridge}
+	del_addrs ${netdev}
     fi
 
     # Attach the real interface to the bridge.
--- xen-2.0/tools/examples/network.orig	2005-06-13 14:27:19.000000000 +0200
+++ xen-2.0/tools/examples/network	2005-06-13 14:36:01.000000000 +0200
@@ -79,7 +79,7 @@
     local src=$1
     ip addr show dev ${src} | egrep '^ *inet ' | sed -e "
 s/inet/ip addr del/
-s@\([0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+\)/[0-9]\+@\1@
+s@\([0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+\)/[0-9]\*@\1@
 s/${src}/dev ${src}/
 " | sh -e
 }
