diff -dur -x '*~' xen-3.0.2.orig/tools/examples/block xen-3.0.2/tools/examples/block
--- xen-3.0.2.orig/tools/examples/block	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/block	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 dir=$(dirname "$0")
 . "$dir/block-common.sh"
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/block-enbd xen-3.0.2/tools/examples/block-enbd
--- xen-3.0.2.orig/tools/examples/block-enbd	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/block-enbd	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # Usage: block-enbd [bind server ctl_port |unbind node]
 #
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/block-nbd xen-3.0.2/tools/examples/block-nbd
--- xen-3.0.2.orig/tools/examples/block-nbd	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/block-nbd	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # Usage: block-nbd [bind server ctl_port |unbind node]
 #
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/network-bridge xen-3.0.2/tools/examples/network-bridge
--- xen-3.0.2.orig/tools/examples/network-bridge	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/network-bridge	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # Default Xen network start/stop script.
 # Xend calls a network script when it starts.
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/network-nat xen-3.0.2/tools/examples/network-nat
--- xen-3.0.2.orig/tools/examples/network-nat	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/network-nat	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # Default Xen network start/stop script when using NAT.
 # Xend calls a network script when it starts.
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/network-route xen-3.0.2/tools/examples/network-route
--- xen-3.0.2.orig/tools/examples/network-route	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/network-route	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # Default Xen network start/stop script.
 # Xend calls a network script when it starts.
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/vif-bridge xen-3.0.2/tools/examples/vif-bridge
--- xen-3.0.2.orig/tools/examples/vif-bridge	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/vif-bridge	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # /etc/xen/vif-bridge
 #
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/vif-nat xen-3.0.2/tools/examples/vif-nat
--- xen-3.0.2.orig/tools/examples/vif-nat	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/vif-nat	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # /etc/xen/vif-nat
 #
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/vif-route xen-3.0.2/tools/examples/vif-route
--- xen-3.0.2.orig/tools/examples/vif-route	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/vif-route	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #============================================================================
 # /etc/xen/vif-route
 #
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/vtpm xen-3.0.2/tools/examples/vtpm
--- xen-3.0.2.orig/tools/examples/vtpm	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/vtpm	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 dir=$(dirname "$0")
 . "$dir/vtpm-hotplug-common.sh"
diff -dur -x '*~' xen-3.0.2.orig/tools/examples/vtpm-delete xen-3.0.2/tools/examples/vtpm-delete
--- xen-3.0.2.orig/tools/examples/vtpm-delete	2006-04-10 00:05:53.000000000 +0200
+++ xen-3.0.2/tools/examples/vtpm-delete	2006-04-30 20:18:41.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # This scripts must be called the following way:
 # vtpm-delete <domain name>
