--- xen-3.0.4_1-src/tools/check/check_python_devel~	2008-05-24 00:01:14.000000000 +0300
+++ xen-3.0.4_1-src/tools/check/check_python_devel	2008-05-24 00:01:16.695654488 +0300
@@ -5,9 +5,9 @@
 
 python -c '
 import os.path, sys
-for p in sys.path:
-	if os.path.exists(p + "/config/Makefile"):
-		sys.exit(0)
+p = os.path.join(sys.prefix, sys.lib, "python%s" % sys.version[:3], "config", "Makefile")
+if os.path.exists(p):
+	sys.exit(0)
 sys.exit(1)
 ' || RC=1 
 
