--- xen-4.15.1/tools/Makefile.orig	2021-11-09 17:01:44.785575980 +0100
+++ xen-4.15.1/tools/Makefile	2021-11-09 20:32:30.967065733 +0100
@@ -237,6 +237,7 @@ subdir-all-qemu-xen-dir: qemu-xen-dir-fi
 		$$enable_trace_backend \
 		--prefix=$(prefix) \
 		--libdir=$(LIBEXEC_LIB) \
+		--libexecdir=$(LIBEXEC_LIB) \
 		--includedir=$(LIBEXEC_INC) \
 		--extra-cflags="-DXC_WANT_COMPAT_EVTCHN_API=1 \
 		-DXC_WANT_COMPAT_GNTTAB_API=1 \
