diff -dur xen-4.2.0.orig/tools/libfsimage/common/Makefile xen-4.2.0/tools/libfsimage/common/Makefile
--- xen-4.2.0.orig/tools/libfsimage/common/Makefile	2012-09-17 12:21:18.000000000 +0200
+++ xen-4.2.0/tools/libfsimage/common/Makefile	2012-10-23 09:20:47.000000000 +0200
@@ -5,7 +5,7 @@
 MINOR = 0
 
 LDFLAGS-$(CONFIG_SunOS) = -Wl,-M -Wl,mapfile-SunOS
-LDFLAGS-$(CONFIG_Linux) = -Wl,mapfile-GNU
+LDFLAGS-$(CONFIG_Linux) = -Wl,mapfile-GNU -ldl
 LDFLAGS = $(LDFLAGS-y)
 
 CFLAGS += $(PTHREAD_CFLAGS)
diff -dur xen-4.2.0.orig/tools/libxl/Makefile xen-4.2.0/tools/libxl/Makefile
--- xen-4.2.0.orig/tools/libxl/Makefile	2012-09-17 12:21:18.000000000 +0200
+++ xen-4.2.0/tools/libxl/Makefile	2012-10-23 09:24:52.000000000 +0200
@@ -173,8 +173,8 @@
 libxlutil.so.$(XLUMAJOR): libxlutil.so.$(XLUMAJOR).$(XLUMINOR)
 	ln -sf $< $@
 
-libxlutil.so.$(XLUMAJOR).$(XLUMINOR): $(LIBXLU_OBJS)
-	$(CC) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxlutil.so.$(XLUMAJOR) $(SHLIB_LDFLAGS) -o $@ $^ $(LIBXLU_LIBS) $(APPEND_LDFLAGS)
+libxlutil.so.$(XLUMAJOR).$(XLUMINOR): $(LIBXLU_OBJS) libxenlight.so
+	$(CC) $(LDFLAGS) -Wl,$(SONAME_LDFLAG) -Wl,libxlutil.so.$(XLUMAJOR) $(SHLIB_LDFLAGS) -o $@ $^ $(LIBXLU_LIBS) -L. -lxenlight $(APPEND_LDFLAGS)
 
 libxlutil.a: $(LIBXLU_OBJS)
 	$(AR) rcs libxlutil.a $^
