diff -dur xen-4.2.0.orig/tools/misc/xenpvnetboot xen-4.2.0/tools/misc/xenpvnetboot
--- xen-4.2.0.orig/tools/misc/xenpvnetboot	2012-09-17 12:21:18.000000000 +0200
+++ xen-4.2.0/tools/misc/xenpvnetboot	2012-10-22 10:46:30.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #
 # Copyright (C) 2010 Oracle. All rights reserved.
 #
diff -dur xen-4.2.0.orig/tools/qemu-xen/scripts/analyse-9p-simpletrace.py xen-4.2.0/tools/qemu-xen/scripts/analyse-9p-simpletrace.py
--- xen-4.2.0.orig/tools/qemu-xen/scripts/analyse-9p-simpletrace.py	2012-09-10 20:10:52.000000000 +0200
+++ xen-4.2.0/tools/qemu-xen/scripts/analyse-9p-simpletrace.py	2012-10-22 10:46:30.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 # Pretty print 9p simpletrace log
 # Usage: ./analyse-9p-simpletrace <trace-events> <trace-pid>
 #
diff -dur xen-4.2.0.orig/tools/qemu-xen/scripts/simpletrace.py xen-4.2.0/tools/qemu-xen/scripts/simpletrace.py
--- xen-4.2.0.orig/tools/qemu-xen/scripts/simpletrace.py	2012-09-10 20:10:52.000000000 +0200
+++ xen-4.2.0/tools/qemu-xen/scripts/simpletrace.py	2012-10-22 10:46:30.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #
 # Pretty-printer for simple trace backend binary trace files
 #
